### PR TITLE
Fixing and udpating the GDPR procedure

### PR DESCRIPTION
**Pull request by MichalMaler**
Signed-off-by: Michal Maléř <mmaler@redhat.com>

* Fixing and updating the GDPR procedure.
* Applying macros for proper asciidoc translation
* Removing `italics` and `<>` from antora_playbook attribute values @rkratky

### CODE REVIEW

**Review by vparfonov — APPROVED** [review]

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+{
+ "name": "vparfono",
+ "links": [
+   {
+.
+.
+.
+   }
+ ],
+ "email": "vitaly.parfonov@gmail.com",
```
Comment (vparfonov) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
 "email": "vparfono@redhat.com",
```

File: antora-playbook.yml [hunk 2]
```diff
@@ -94,8 +94,8 @@ asciidoc:
     prod-prev-ver: 7.18
     prod-short: Che
     prod-upstream: Eclipse{nbsp}Che
-    prod-url-secure: +https+://__<che-host>__
-    prod-url: +https+://__<che-host>__:__<che-port>__
+    prod-url-secure: +https+://che-host
```
Comment (rkratky) [inline antora-playbook.yml]:
```suggestion
    prod-url-secure: https://che-host
```

File: antora-playbook.yml [hunk 3]
```diff
@@ -94,8 +94,8 @@ asciidoc:
...
+    prod-url: +https+://che-host:che-port
```
Comment (rkratky) [inline antora-playbook.yml]:
```suggestion
    prod-url: https://che-host:che-port
```
As discussed, this doesn't work in Antora and must be handled by pass-through macros in the text.

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 4]
```diff
@@ -0,0 +1,164 @@
+
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+:parent-context-of-removing-user-data: {context}
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+[id="removing-user-data_{context}"]
+= Removing user data
+
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc:3]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+:context: removing-user-data
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+
+== Removing user data according to GDPR
+
+The General Data Protection Regulation (GDPR) law enforces the right for individuals to have personal data erased.
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]: Add a link to some source that explains what GDPR is.

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+
+The following procedure describes how to remove a user’s data from a cluster and the {identity-provider} database.
+
+[NOTE]
+====
+* Following commands use the default namespace, `_{prod-namespace}_`, as a user's namespace example after the `-n` option.
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
Following commands use the default namespace, `{prod-namespace}`, as a user's namespace example after the `-n` option.
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+====
+
+.Prerequisites
+
+* A user or an administrator authorization token. If a user wants to delete any other data except the data bound to their user account, the `admin` privileges are required.
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
* A user or an administrator authorization token. To delete any other data except the data bound to a user account, `admin` privileges are required. The `admin` is a special {prod-short} administrator account pre-created and enabled using the `pass:[CHE_SYSTEM_ADMIN__NAME]` and `pass:[CHE_SYSTEM_SUPER__PRIVILEGED__MODE = true]` Custom Resource definitions.
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
++
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+The `admin` is a  special {prod-short} administrator account pre-created and enabled using the:
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
++
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+[subs="+quotes,macros"]
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 17]
```diff
@@ -0,0 +1,164 @@
...
+* `pass:[CHE_SYSTEM_ADMIN__NAME]`
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 18]
```diff
@@ -0,0 +1,164 @@
...
+[subs="+quotes,macros"]
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 19]
```diff
@@ -0,0 +1,164 @@
...
+* `pass:[CHE_SYSTEM_SUPER__PRIVILEGED__MODE = true]`
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 20]
```diff
@@ -0,0 +1,164 @@
...
+
+* A user or an administrator authorization token. If a user wants to delete any other data except the data bound to their user account, the `admin` privileges are required.
++
+The `admin` is a  special {prod-short} administrator account pre-created and enabled using the:
++
+[subs="+quotes,macros"]
+* `pass:[CHE_SYSTEM_ADMIN__NAME]`
+[subs="+quotes,macros"]
+* `pass:[CHE_SYSTEM_SUPER__PRIVILEGED__MODE = true]` 
++
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 21]
```diff
@@ -0,0 +1,164 @@
...
+Custom Resource definitions.
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 22]
```diff
@@ -0,0 +1,164 @@
...
++
+[subs="+quotes,macros"]
+----
+spec:
+ server:
+   customCheProperties:
+     pass:[CHE_SYSTEM_SUPER__PRIVILEGED__MODE]: 'true'
+     pass:[CHE_SYSTEM_ADMIN__NAME]: '__<admin-name>__'
+
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 23]
```diff
@@ -0,0 +1,164 @@
...
+----
+spec:
+ server:
+   customCheProperties:
+     pass:[CHE_SYSTEM_SUPER__PRIVILEGED__MODE]: 'true'
+     pass:[CHE_SYSTEM_ADMIN__NAME]: '__<admin-name>__'
+     
+----
++
+If needed, use commands below for creating an `admin` user:
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
If needed, use commands below for creating the `admin` user:
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 24]
```diff
@@ -0,0 +1,164 @@
...
+  --type merge \
+  -p '{ "spec": { "server": {"customCheProperties": pass:[{"CHE_SYSTEM_SUPER__PRIVILEGED__MODE": "true"}] } }}' \
+  -n __{prod-namespace}__
+----
++
+[subs="+quotes,macros,attributes"]
+----
+$ oc patch checluster {prod-checluster} \
+  --type merge \
+  pass:[-p '{ "spec": { "server": {"customCheProperties": {"CHE_SYSTEM_ADMIN__NAME": "__<admin-name>__"}] } }}' \
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]: The pass-through macro must be used only the things you don't want interpreted. Your code results in this:
image
(I.e. review locally before pushing :)
```suggestion
  -p '{ "spec": { "server": {"customCheProperties": {"pass:[CHE_SYSTEM_ADMIN__NAME]": "__<admin-name>__"} } }}' \
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 25]
```diff
@@ -0,0 +1,164 @@
...
+     
+----
++
+If needed, use commands below for creating an `admin` user:
++
+[subs="+quotes,macros,attributes"]
+----
+$ oc patch checluster {prod-checluster} \
+  --type merge \
+  -p '{ "spec": { "server": {"customCheProperties": pass:[{"CHE_SYSTEM_SUPER__PRIVILEGED__MODE": "true"}] } }}' \
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
  -p '{ "spec": { "server": {"customCheProperties": {"pass:[CHE_SYSTEM_SUPER__PRIVILEGED__MODE]": "true"} } }}' \
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 26]
```diff
@@ -0,0 +1,164 @@
...
+ 
+** To find a user by email: 
++
+[subs="+quotes,macros,attributes"]
+----
+$ curl -X GET \
+  --header 'Authorization: Bearer __<user-token>__' \
+  'pass:c,m,a,q[+https:+//__<{prod-host}>__]/api/user/find?email=__<email>__'
+----
+
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
+
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 27]
```diff
@@ -0,0 +1,164 @@
...
+.Example of obtaining a user ID
+
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
+
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 28]
```diff
@@ -0,0 +1,164 @@
...
+This example uses `vparfono` as a local user name.
+
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
+
```

**Review by ScrewTSW — APPROVED** [review]
LGTM

File: modules/administration-guide/partials/proc_removing-user-data.adoc (hunk 1 above)
Comment (MichalMaler) [reply modules/administration-guide/partials/proc_removing-user-data.adoc]: will edit

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 29]
```diff
@@ -0,0 +1,164 @@
...
+ ],
+ "email": "vitaly.parfonov@gmail.com",
+ "id": "921b6f33-2657-407e-93a6-fb14cf2329ce"
+}
+----
+====
+
+.Procedure
+
+. Update the `customresourceCR` file to permit the removal of a user’s data from the {identity-provider} database:
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]: Is the file really called `customresourceCR`?

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 30]
```diff
@@ -0,0 +1,164 @@
...
++
+[subs="+quotes,macros,attributes"]
+----
+$ oc patch checluster/{prod-checluster} \
+  --patch "{\"spec\":{\"server\":{\"customCheProperties\": pass:[{\"CHE_KEYCLOAK_CASCADE__USER__REMOVAL__ENABLED\"]: \"true\"}}}}" \
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
  --patch "{\"spec\":{\"server\":{\"customCheProperties\": {\"pass:[CHE_KEYCLOAK_CASCADE__USER__REMOVAL__ENABLED]\": \"true\"}}}}" \
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 31]
```diff
@@ -0,0 +1,164 @@
...
+  --type=merge -n __{prod-namespace}__
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
  --type=merge -n _{prod-namespace}_
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 32]
```diff
@@ -0,0 +1,164 @@
...
+----
+$ curl -i -X DELETE \
+  --header 'Authorization: Bearer __<user-token>__' \
+  pass:c,m,a,q[+https:+//__<{prod-host}>__]/api/user/__<user-id>__
+----
+
+
+.Verification
+
+The execution of the `curl -i -X DELETE --header 'Authorization: Bearer __<user-token>__' pass:c,m,a,q[+https:+//__<{prod-host}>__]/api/user/__<user-id>__` command, returns the code 204 as the API response.
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
Running the following command returns code `204` as the API response:
+
 [subs="+quotes,macros,attributes"]
 ====
$ curl -i -X DELETE \
  --header 'Authorization: Bearer __<user-token>__' \
  pass:c,m,a,q[+https:+//__<{prod-host}>__]/api/user/__<user-id>__
====
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 33]
```diff
@@ -0,0 +1,164 @@
...
+
+
+.Additional resources
+To remove the data of all the users, follow the instructions for xref:installation-guide:uninstalling-che.adoc[].
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
+
To remove the data of all users, follow the instructions for xref:installation-guide:uninstalling-che.adoc[].
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 34]
```diff
@@ -0,0 +1,164 @@
...
+
+
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc:151]:
```suggestion
```

File: modules/administration-guide/partials/proc_removing-user-data.adoc [hunk 35]
```diff
@@ -0,0 +1,164 @@
...
+:context: {parent-context-of-removing-user-data}
```
Comment (rkratky) [inline modules/administration-guide/partials/proc_removing-user-data.adoc]:
```suggestion
```

**Review by rkratky — CHANGES_REQUESTED** [review]

File: modules/administration-guide/partials/proc_removing-user-data.adoc (hunk 29 above)
Comment (MichalMaler) [reply modules/administration-guide/partials/proc_removing-user-data.adoc]: Updated to:

. Update the `{prod-checluster}` `CheCluster Custom` Resource (CR) definition to permit the removal of a user’s data from the {identity-provider} database:

File: modules/administration-guide/partials/proc_removing-user-data.adoc (hunk 10 above)
Comment (MichalMaler) [reply modules/administration-guide/partials/proc_removing-user-data.adoc]: Done. Updated with the main GDPR Wiki link.

**Review by rkratky — APPROVED** [review]

**Pull request merged**